Assets/Scripts/Game/System/CameraController.cs

```diff
@@ -4,58 +4,104 @@
+        ClampCameraPosition();
+    }
+
+    private void Zoom(float sizeDelta)
+    {
+        // Clamp input
+        float currentCameraSize = Camera.main.orthographicSize;
+        float targetCameraSize = currentCameraSize + sizeDelta;
+        float clampedTargetCameraSize = Mathf.Clamp(targetCameraSize, minCameraSize, maxCameraSize);
+        float clampedSizeDelta = clampedTargetCameraSize - currentCameraSize;
+
+        // If zooming in, move camera according to desired zoom amount and mouse position
+        if (clampedSizeDelta < 0)
         {
-            // Update zoom momentum
-            lastScrollVelocity = scrollVelocity;
-            scrollDampTimer = 0;
-            
-            // Update mouse position
-            lastMouseViewportPos = Camera.main.ScreenToViewportPoint(Input.mousePosition);
-            lastMouseWorldPos = Camera.main.ScreenToWorldPoint(Input.mousePosition);
+            Vector3 mousePosition = Camera.main.ScreenToWorldPoint(Input.mousePosition);
+            float changeRatio = clampedSizeDelta / Camera.main.orthographicSize;
+            Vector2 cameraDelta = changeRatio * (mousePosition - transform.position) * -1;
+            MoveCamera(cameraDelta.x, cameraDelta.y);
         }
-        else
-        {
-            // Update timer
-            scrollDampTimer += Time.deltaTime;
 
-            // Dampen scrolling
-            scrollVelocity = Mathf.Lerp(lastScrollVelocity, 0, scrollDampTimer / scrollDampTime);
-        }
+        // Zoom camera
+        Camera.main.orthographicSize += clampedSizeDelta;
+    }
+
+    private void MoveCameraUsingMouse()
+    {
+        Vector2 mousePosViewPort = Camera.main.ScreenToViewportPoint(Input.mousePosition);
+        int xDir = mousePosViewPort.x < panScreenEdgeProportion ? -1 : (mousePosViewPort.x > 1 - panScreenEdgeProportion ? 1 : 0);
+        int yDir = mousePosViewPort.y < panScreenEdgeProportion ? -1 : (mousePosViewPort.y > 1 - panScreenEdgeProportion ? 1 : 0);
+
+        xEdgeSpeedManager.UpdateSpeed(panAcceleration * xDir);
+        yEdgeSpeedManager.UpdateSpeed(panAcceleration * yDir);
 
-        // Zoom the camera
-        Camera.main.orthographicSize = Mathf.Clamp(Camera.main.orthographicSize + scrollVelocity * Time.deltaTime, minZoom, maxZoom);
+        MoveCamera(xEdgeSpeedManager.Speed, yEdgeSpeedManager.Speed);
+    }
+
+    private void MoveCamera(float xAddition, float yAddition)
+    {
+        transform.position += new Vector3(xAddition, yAddition, 0);
+    }
+
+    private void ClampCameraPosition()
+    {
+        float cameraSize = Camera.main.orthographicSize;
+        float minCameraSizeX = minCameraSize * Camera.main.aspect;
+        float maxZoomLevel = maxCameraSize - minCameraSize;
+        float zoomLevel = maxCameraSize - cameraSize;
+        float overallMaxDistFromCenterX = worldSizeX - minCameraSizeX + panSideMargins;
+        float currentMaxDistFromCenterX = (zoomLevel / maxZoomLevel) * overallMaxDistFromCenterX;
```

Comment:
This is a fairly complex function. To clarify the meaning (and shorten some lines), I would suggest some clean-ups:

```suggestion
        // Calculate y bounds
        float cameraSize = Camera.main.orthographicSize;
        float boundY = worldHeight / 2 - cameraSize + panSideMargin;

        // Calculate zoom level
        float maxZoomLevel = maxCameraSize - minCameraSize;
        float zoomLevel = maxCameraSize - cameraSize;

        // Calculate x bounds based on zoom level
        float minCameraSizeX = minCameraSize * Camera.main.aspect;
        float maxBoundX = worldWidth / 2 - minCameraSizeX + panSideMargin;
        float boundX = (zoomLevel / maxZoomLevel) * maxBoundX;
        
        // Calculate clamped position values
        float clampedX = Mathf.Clamp(transform.position.x, -boundX, boundX);
        float clampedY = Mathf.Clamp(transform.position.y, -boundY, boundY);
```

Reply:
Committed suggestion.